pypiserver/core.py

```diff
@@ -246,6 +250,15 @@ def _listdir(root):
                               fn=fn, root=root,
                               relfn=fn[len(root) + 1:])
 
+try:
+    from .cache import listdir_cache
+
+    def listdir(root):
+        return listdir_cache.get(root, _listdir)
+except ImportError:
+    listdir = _listdir
```

Comment:
Ah ha. I know what happened. This is older code from my version of core.py that got patched. When I rebased it got pulled back in. 

See this is why you shouldn't game and code.

I'll refactor and redo the PR. Let's just get this version out for now.